applications/GeoMechanicsApplication/python_scripts/geomechanics_analysis.py

```diff
@@ -102,50 +102,57 @@ class GeoMechanicsAnalysis(GeoMechanicsAnalysisBase):
 
+        self.reduction_factor    = project_parameters["solver_settings"]["reduction_factor"].GetDouble()
+        self.increase_factor     = project_parameters["solver_settings"]["increase_factor"].GetDouble()
+        self.min_iterations      = project_parameters["solver_settings"]["min_iterations"].GetInt()
         self.max_delta_time_factor = 1000.0
         if project_parameters["solver_settings"]["time_stepping"].Has("max_delta_time_factor"):
-            self.max_delta_time_factor = \
-                project_parameters["solver_settings"]["time_stepping"]["max_delta_time_factor"].GetDouble()
-
+            self.max_delta_time_factor = project_parameters["solver_settings"]["time_stepping"]["max_delta_time_factor"].GetDouble()
         self.max_delta_time      = self.delta_time * self.max_delta_time_factor
-        self.max_iterations      = project_parameters["solver_settings"]["max_iterations"].GetInt()
-        self.min_iterations      = project_parameters["solver_settings"]["min_iterations"].GetInt()
         self.number_cycles       = project_parameters["solver_settings"]["number_cycles"].GetInt()
+
+        self.max_iterations      = project_parameters["solver_settings"]["max_iterations"].GetInt()
         self.solution_type       = project_parameters["solver_settings"]["solution_type"].GetString()
         self.reset_displacements = project_parameters["solver_settings"]["reset_displacements"].GetBool()
-        self.start_time          = project_parameters["solver_settings"]["start_time"].GetDouble()
-        self.end_time            = project_parameters["problem_data"]["end_time"].GetDouble()
         self.rebuild_level       = project_parameters["solver_settings"]["rebuild_level"].GetInt()
 
     def FinalizeSolutionStep(self):
         super().FinalizeSolutionStep()
 
-        if(self._GetSolver().GetComputingModelPart().ProcessInfo[KratosMultiphysics.NL_ITERATION_NUMBER] > self.max_iterations):
+        if (self._GetSolver().GetComputingModelPart().ProcessInfo[KratosMultiphysics.NL_ITERATION_NUMBER] > self.max_iterations):
             raise Exception("max_number_of_iterations_exceeded")
 
     def RunSolutionLoop(self):
         """This function executes the solution loop of the AnalysisStage
         It can be overridden by derived classes
         """
+
+        # store total displacement field for reset_displacements
         if self._GetSolver().settings["reset_displacements"].GetBool():
             old_total_displacements = [node.GetSolutionStepValue(KratosGeo.TOTAL_DISPLACEMENT)
                                        for node in self._GetSolver().GetComputingModelPart().Nodes]
 
         self._GetSolver().solver.SetRebuildLevel(self.rebuild_level)
 
         while self.KeepAdvancingSolutionLoop():
+            # check against max_delta_time should only be necessary when trying to scale up
             if (self.delta_time > self.max_delta_time):
                 self.delta_time = self.max_delta_time
```

Comment:
```suggestion
            self.delta_time = min(self.delta_time, self.max_delta_time)
```

Reply:
That would imply that we unconditionally get the logging about "reducing delta_time to max_delta_time"
even when that doesn't happen.